lib/ts/recipe/emailpassword/utils.ts

```diff
@@ -114,25 +119,58 @@ function validateAndNormaliseResetPasswordUsingTokenConfig(
     };
 }
 
-function normaliseSignInFormFields(formFields: NormalisedFormField[]) {
-    return formFields
-        .filter((filter) => filter.id === FORM_FIELD_EMAIL_ID || filter.id === FORM_FIELD_PASSWORD_ID)
-        .map((field) => {
-            return {
-                id: field.id,
-                // see issue: https://github.com/supertokens/supertokens-node/issues/36
-                validate: field.id === FORM_FIELD_EMAIL_ID ? field.validate : defaultValidator,
-                optional: false,
-            };
+function normaliseSignInFormFields(formFields?: TypeInputFormField[]) {
+    let normalisedFormFields: NormalisedFormField[] = [];
+    if (formFields !== undefined) {
+        formFields.forEach((field) => {
+            if (field.id === FORM_FIELD_PASSWORD_ID) {
+                normalisedFormFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultPasswordValidator : field.validate,
```

Comment:
In general, you shouldn't have a validator for the password field during sign-in, since we'd want to check the DB in most cases anyway to prevent cases where a password validation change is preventing people from logging in. I don't mind giving people the option to define this specifically, but by default it should let everything pass.

```suggestion
                    validate: field.validate === undefined ? defaultValidator : field.validate,
```